src/OmniSharp.Cake/Extensions/ResponseExtensions.cs

```diff
@@ -85,6 +86,44 @@ public static Task<QuickFixResponse> TranslateAsync(this QuickFixResponse respon
             return response;
         }
 
+        public static async Task<RunCodeActionResponse> TranslateAsync(this RunCodeActionResponse response, OmniSharpWorkspace workspace)
+        {
+            if (response?.Changes == null)
+            {
+                return response;
+            }
+
+            var fileOperations = new List<FileOperationResponse>();
+            var changes = new Dictionary<string, List<LinePositionSpanTextChange>>();
+
+            foreach (var fileOperation in response.Changes)
+            {
+                if (fileOperation.ModificationType == FileModificationType.Modified &&
+                    fileOperation is ModifiedFileResponse modifiedFile)
+                {
+                    await PopulateModificationsAsync(modifiedFile, workspace, changes);
+                }
+
+                fileOperations.Add(fileOperation);
+            }
+
+            foreach (var change in changes)
+            {
+
+                if (!(fileOperations.FirstOrDefault(x => x.FileName == change.Key &&
+                                                         x.ModificationType == FileModificationType.Modified)
+                    is ModifiedFileResponse modifiedFile))
+                {
+                    continue;
+                }
+
+                modifiedFile.Changes = change.Value;
```

Comment:
This means that `modifiedFile.Changes` will only ever have the last matching value. Is that intentional or should there by some sort of merging?

Reply:
No, `change` is of type `KeyValuePair<string, List<LinePositionSpanTextChange>>` and the dictionary it's contained in is populated above, using the `PopulateModificationsAsync`.

This is the same thing we do with rename responses also. it's ugly, but due to the fact that loaded scripts are "merged" into the base script we might end up in a situation where a `ModifiledFileResponse` contains modifications belonging to a different file. This is what we are trying to fix in this method, e.g. loop through all modifications, map line numbers and assign them to the correct `ModifiedFileResponse`.

Someday, Cake should not "copy & paste" loaded script contents in the base file and instead use `#load` as provided by Roslyn scripting.

Reply:
got it. Thanks!

Reply:
@filipw I see somebody has already written a blog post that will help me getting started :wink: https://www.strathweb.com/2016/06/implementing-custom-load-behavior-in-roslyn-scripting/